data/transactions/logic/eval.go

```diff
@@ -74,54 +74,43 @@ type stackValue struct {
 	Bytes []byte
 }
 
-func (sv *stackValue) argType() StackType {
+func (sv stackValue) argType() StackType {
```

Comment:
@jannotti Similar to my question about `debugState`, can you help me understand how you decided to change `stackValue` interface method definitions from pointer to value receivers?

From my review, I'm _guessing_ the change tries to signal that `stackValue` is considered immutable-ish.

Reply:
I think I missed the first because of it being in a weird place.  I'll fix and relocate.

stackValueFromTealValue could be changed but I didn't notice it.  My bias is to use values for read-only if the value in question is, say, 32 bytes or smaller.  That would include basics.TealValue, so I suppose I'd prefer a value argument here.

It would be nice to have a well-written benchmark to check my intuition on that 32 byte size.

Reply:
Changed both

Reply:
@jannotti Thanks for the change in https://github.com/algorand/go-algorand/pull/3612/commits/51a882c2f40d9acf13565e5713a71c064b814b18.  And got it RE:  heuristics for considering value vs pointer arguments/receivers.